core/block_svg.ts

```diff
@@ -600,17 +601,54 @@ export class BlockSvg
     return menuOptions;
   }
 
+  /**
+   * Gets the location in which to show the context menu for this block.
+   * Use the location of a click if the block was clicked, or a location
+   * based on the block's fields otherwise.
+   */
+  protected calculateContextMenuLocation(e: Event): Coordinate {
+    // Open the menu where the user clicked, if they clicked
+    if (e instanceof PointerEvent) {
+      return new Coordinate(e.clientX, e.clientY);
+    }
+
+    // Otherwise, calculate a location.
+    // Get the location of the top-left corner of the block in
+    // screen coordinates.
+    const blockCoords = svgMath.wsToScreenCoordinates(
+      this.workspace,
+      this.getRelativeToSurfaceXY(),
+    );
+
+    // Prefer a y position below the first field in the block.
+    const fieldBoundingClientRect = this.inputList
+      .filter((input) => input.isVisible())
+      .flatMap((input) => input.fieldRow)
+      .filter((f) => f.isVisible())[0]
+      ?.getSvgRoot()
+      ?.getBoundingClientRect();
+
+    const y =
+      fieldBoundingClientRect && fieldBoundingClientRect.height
+        ? fieldBoundingClientRect.y + fieldBoundingClientRect.height
+        : blockCoords.y + this.height;
+
+    return new Coordinate(blockCoords.x + 5, y + 5);
```

Comment:
Does the +5 offset need to be reversed for RTL?

Reply:
Good catch. I thought I would have to give it the right edge of the block instead of the left (so calculating the x + width), but it works without doing that, and it is too far to the right if I do try to do that. I guess something else is already taking into account the RTL and making this work. But I did need to subtract 5 instead.